sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -1996,6 +1998,461 @@ internal virtual void SetNameFieldsIfNull()
+                            .ConfigureAwait(false);
+
+                        continuationToken = jsonResponse.Value.Continuation;
+                        using (var document = JsonDocument.Parse(jsonResponse.Value.Body))
+                        {
+                            var response = DeserializeSetAccessControlRecursiveResponse(document.RootElement);
+                            int currentDirectoriesSuccessfulCount = response.DirectoriesSuccessful ?? 0;
+                            int currentFilesSuccessfulCount = response.FilesSuccessful ?? 0;
+                            int currentFailureCount = response.FailureCount ?? 0;
+                            if (progressHandler != null)
+                            {
+                                var failedEntries = response.FailedEntries
+                                    .Select(failedEntry => new ChangeAccessControlListResultFailedEntry()
+                                    {
+                                        Name = failedEntry.Name,
+                                        Type = failedEntry.Type,
+                                        ErrorMessage = failedEntry.ErrorMessage,
+                                    }).ToList();
+                                progressHandler.Report(new ChangeAccessControlListPartialResult()
+                                {
+                                    DirectoriesSuccessfulCount = currentDirectoriesSuccessfulCount,
+                                    FilesSuccessfulCount = currentFilesSuccessfulCount,
+                                    FailureCount = failureCount,
+                                    FailedEntries = failedEntries,
+                                });
+                            }
+                            directoriesSuccessfulCount += currentDirectoriesSuccessfulCount;
+                            filesSuccessfulCount += currentFilesSuccessfulCount;
+                            failureCount += currentFailureCount;
+                        }
+
+                        cancellationToken.ThrowIfCancellationRequested();
+                    } while (!string.IsNullOrEmpty(continuationToken));
+
+                    return new ChangeAccessControlListResult()
+                        {
+                            DirectoriesSuccessfulCount = directoriesSuccessfulCount,
+                            FilesSuccessfulCount = filesSuccessfulCount,
+                            FailureCount = failureCount,
+                        };
+                }
+                catch (Exception ex)
+                {
+                    Pipeline.LogException(ex);
+                    throw;
+                }
+                finally
+                {
+                    Pipeline.LogMethodExit(nameof(DataLakePathClient));
+                }
+            }
+        }
+
+        private static SetAccessControlRecursiveResponse DeserializeSetAccessControlRecursiveResponse(JsonElement element)
```

Comment:
If we have to end up keeping the json work around, let's move this to a different file.  Maybe ```DataLakeExtensions.cs```?

Reply:
moved.